hwy/contrib/sort/order-emulate-inl.h

```diff
@@ -0,0 +1,224 @@
+    using TI = MakeSigned<T>;
+    constexpr int kMSB = 8 * sizeof(T) - 1;
+    constexpr T neg_flip = T((T(1) << kMSB) - 1); 
+    const T a_neg = 0 - (a >> kMSB);
+    const T b_neg = 0 - (b >> kMSB);
+    // Signed-domain keys (xor 0x7FFF.. only for negatives)
+    const T sa = a ^ (a_neg & neg_flip);
+    const T sb = b ^ (b_neg & neg_flip);
+    return static_cast<TI>(sa) < static_cast<TI>(sb);
+  }
+  template <class D>
+  HWY_INLINE Mask<D> Compare(D, Vec<D> a, Vec<D> b) const {
+    // specialized less than, -0.0 < +0.0, and NaNs are not ordered
+    return LtBin(a, b);
+  }
+
+  // Two halves of Sort2, used in ScanMinMax.
+  template <class D>
+  HWY_INLINE Vec<D> First(D /* tag */, const Vec<D> a, const Vec<D> b) const {
+    return IfThenElse(LtBin(a, b), a, b);
+  }
+
+  template <class D>
+  HWY_INLINE Vec<D> Last(D /* tag */, const Vec<D> a, const Vec<D> b) const {
+    return IfThenElse(LtBin(a, b), b, a);
+  }
+
+  template <class D>
+  HWY_INLINE Vec<D> FirstOfLanes(D d, Vec<D> v,
+                                 T* HWY_RESTRICT /* buf */) const {
+    const RebindToSigned<D> di;
+    using VI = Vec<decltype(di)>;
+    VI key = LtBinKey(v);
+    VI min = MinOfLanes(di, key);
+    Mask<D> m = RebindMask(d, Eq(min, key));
+    return MaxOfLanes(d, IfThenElseZero(m, v));
+  }
+
+  template <class D>
+  HWY_INLINE Vec<D> LastOfLanes(D d, Vec<D> v,
+                                T* HWY_RESTRICT /* buf */) const {
+    const RebindToSigned<D> di;
+    using VI = Vec<decltype(di)>;
+    VI key = LtBinKey(v);
+    VI max = MaxOfLanes(di, key);
+    Mask<D> m = RebindMask(d, Eq(max, key));
+    return MaxOfLanes(d, IfThenElseZero(m, v));
+  }
+
+  template <class D>
+  HWY_INLINE Vec<D> FirstValue(D d) const {
+    return Set(d, ExponentMask<TF>() | SignMask<TF>());
+  }
+
+  template <class D>
+  HWY_INLINE Vec<D> LastValue(D d) const {
+    return Set(d, ExponentMask<TF>());
+  }
+
+  // Returns the next distinct smaller value unless already -inf.
+  template <class D, class V = Vec<D>>
+  HWY_INLINE V PrevValue(D, V v) const {
+    return NextSortValueBits<true>(v);
+  }
+
+  // Next representable value in total order by ±1 ULP, saturating at ±Inf.
+  //   IsDown = false → next larger
+  //   IsDown = true  → next smaller
+  template <bool IsDown, class V>
+  HWY_INLINE V NextSortValueBits(V u) const {
+    const DFromV<V> d;
+    using M = Mask<decltype(d)>;
+    constexpr T kSignBit = SignMask<T>();
+    constexpr T kBoundaryUp = SignMask<T>() - 1;
+    const V sign_bit = Set(d, kSignBit);
+    const V all1 = Set(d, T(~T(0)));
+    const V one = Set(d, T(1));
+    // Detect saturation at ±Inf
+    const M is_target_inf = Eq(u, IsDown ? FirstValue(d) : LastValue(d));
+    // Transform to monotonic space: flip sign for positives, invert for negatives
+    const M is_neg = TestBit(u, sign_bit);
+    const V key = Xor(u, IfThenElse(is_neg, all1, sign_bit));
+    // Boundary detection: +0/-0 swap needs a step of 2 instead of 1
```

Comment:
Do we really need -0 != 0? Wouldn't native float comparison consider them equal/equivalent?